pkg/client/query/sessionquerier.go

```diff
@@ -16,8 +18,11 @@ var _ client.SessionQueryClient = (*sessionQuerier)(nil)
 // querying of onchain session information through a single exposed method
 // which returns an sessiontypes.Session struct
 type sessionQuerier struct {
-	clientConn     grpc.ClientConn
-	sessionQuerier sessiontypes.QueryClient
+	clientConn        grpc.ClientConn
+	sessionQuerier    sessiontypes.QueryClient
+	sharedQueryClient client.SharedQueryClient
+	sessionsCache     KeyValueCache[*sessiontypes.Session]
```

Comment:
Why is this the only pointer?

Reply:
The reason is that the `sessionQueryClient` returns a `QueryGetSessionResponse` which has a `Session` pointer

```go
type QueryGetSessionResponse struct {
	Session *Session `protobuf:"bytes,1,opt,name=session,proto3" json:"session,omitempty"`
}
```

I tried to align the cache type with whatever the querier returns.

Reply:
:+1: